### PR TITLE
GVM 22.4

### PR DESCRIPTION
**Pull request by MocioF**
This PR adds Greenbone Vulnerability Management version 22.4 to gentoo and fixes some bugs found in previous version.

### CODE REVIEW

File: net-analyzer/openvas-scanner/files/openvas.conf
Comment (Flowdalic) [inline net-analyzer/openvas-scanner/files/openvas.conf:1]: This either requires a new file name or that we immediately drop openvas-scanner-21.4.4.-r2. Otherwise, the -r2 ebuild would install a different file depending on the time of installation.

Comment (MocioF) [reply net-analyzer/openvas-scanner/files/openvas.conf:1]: It was wrong even for 21.4.
https://wiki.gentoo.org/wiki/Greenbone_Vulnerability_Management#Redis

Comment (MocioF) [reply net-analyzer/openvas-scanner/files/openvas.conf:1]: Yes, but really the /etc/redis/redis.conf file inserted by dev-db/redis-7.0.11 has Unix socket lines commented.
The wiki says hot to modify it, even if in the wiki page there is a wrong reference to /etc/redis.conf  https://wiki.gentoo.org/wiki/Greenbone_Vulnerability_Management#Redis

If I undertand, you think it is better to use openvas-scanner's ebuild to add a 
an openvas-redis.conf
a  openvas-redis.service for systemd
a  openvas-redis.confd for openrc
to run a separate redis instance for gvm

I will try this approach to write a new service to start a redis instance dedicated to openvas.
However, I think it is better to leave the modification in openvas.conf for 21.4.4-r3 and to drop 21.4.4-r2 (it doesn't build).

Comment (MocioF) [reply net-analyzer/openvas-scanner/files/openvas.conf:1]: Hi @Flowdalic I have made some changes to use a separate instance of redis for openvas. Now I am making some tests before commit.
I have a question: do you think it would be good to create a systemd service to start all GVM service at once (notus-scanner, ospd-openvas, gvmd and gsad)?

File: dev-db/pg-gvm/pg-gvm-22.4.0.ebuild
Comment (Flowdalic) [inline dev-db/pg-gvm/pg-gvm-22.4.0.ebuild:53]: I am not sure how helpful this message is. And I believe this should be displayed at most once and not on every update of the package. Therefore, please consider using `readme.gentoo-r1.eclass`.

File: dev-db/pg-gvm/pg-gvm-22.4.0.ebuild
Comment (Flowdalic) [inline dev-db/pg-gvm/pg-gvm-22.4.0.ebuild:23]: Is `acct-{user,group}/gvm` really a `(R)DEPEND` of pg-gvm?

File: dev-db/pg-gvm/pg-gvm-22.4.0.ebuild
Comment (Flowdalic) [inline dev-db/pg-gvm/pg-gvm-22.4.0.ebuild:37]: Do you really need to explicitly call the `cmake_src_*` functions? I believe you should be able to drop the explicit declaration of `src_{prepare,configure,compile,install}`.

File: dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild
Comment (Flowdalic) [inline dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild:43]: Drop explicit phase definition?

File: dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild
Comment (Flowdalic) [inline dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild:64]: Drop explicit phase definition?

File: dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild
Comment (Flowdalic) [inline dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild:84]: Drop explicit phase definition?

File: net-analyzer/gsad/gsad-21.4.4-r1.ebuild
Comment (Flowdalic) [inline net-analyzer/gsad/gsad-21.4.4-r1.ebuild:45]: What is the source of those patches? Are the already upstreamed? Or is there maybe a pending PR for them?

File: net-analyzer/gvm-libs/Manifest
Comment (Flowdalic) [inline net-analyzer/gvm-libs/Manifest:5]: Why do we need to add four versions of gvm-libs? Would just adding 22.6.1 enough? Or maybe 22.6.1 and 22.5.2? Or maybe 22.6.1 and 22.4.6?

File: profiles/features/musl/package.mask
Comment (Flowdalic) [inline profiles/features/musl/package.mask:61]: I don't see a dependency of notus-scanner "in the previous block"? But maybe I am missing something.
```suggestion
# New packages that require packages masked by the previous block.
```

**Review by Flowdalic — CHANGES_REQUESTED** [review]

File: dev-db/pg-gvm/pg-gvm-22.4.0.ebuild
Comment (MocioF) [reply dev-db/pg-gvm/pg-gvm-22.4.0.ebuild:23]: Not sure.
Here: https://wiki.gentoo.org/wiki/Greenbone_Vulnerability_Management#PostgreSQL_backend
we ask to create the gvm database-user to be the same as the user running gvmd, and to grant all privileges to gvm on the gvmd database.
The extension has to be created on the same database. 

What (I think) the gentoo user has to do after installation is:
~$ sudo -u gvm psql gvmd
gvmd=>	CREATE EXTENSION "pg-gvm";
gvmd=>	\q

File: net-analyzer/gvm-libs/Manifest
Comment (MocioF) [reply net-analyzer/gvm-libs/Manifest:5]: Reading here: https://forum.greenbone.net/t/version-compatibility/14689/2 all versions 22.x will be compatible with 22.4. 
Version 22.4.6 is the first one I added when I started writing the ebuild.
Then later versions were released which I tested and added.
I think we could just leave only version 22.6.1

File: net-analyzer/gsad/gsad-21.4.4-r1.ebuild
Comment (MocioF) [reply net-analyzer/gsad/gsad-21.4.4-r1.ebuild:45]: I have backported this https://github.com/greenbone/gsad/commit/62c4c9180e8f94e0b148dbba15cca9d886079f57
and posted the patch in a br: https://bugs.gentoo.org/905411
I think the patch is necessary if we want to keep version 21.4 in the tree.

I really think we could remove 21.4 because it is obsolete and marked as obsolete in the same GVM scan results, but I'm still waiting for a reply from @j-licht to add me as a proxied maintainer.

File: net-analyzer/openvas-scanner/files/openvas.conf
Comment (MocioF) [reply net-analyzer/openvas-scanner/files/openvas.conf:1]: Hi @Flowdalic, I dropped 21.4.4.-r2, provided a separate redis instance configuration and added a gvm systemd service.
Can you resolve this conversation?

File: net-analyzer/gsad/gsad-21.4.4-r1.ebuild
Comment (thesamesam) [reply net-analyzer/gsad/gsad-21.4.4-r1.ebuild:45]: I think you can go ahead and add yourself.

**Comment by thesamesam** [comment]
Could you squash some of the unnecessary commits (or fix their summaries at least (the first line))?

File: net-analyzer/gsad/gsad-21.4.4-r1.ebuild
Comment (MocioF) [reply net-analyzer/gsad/gsad-21.4.4-r1.ebuild:45]: @thesamesam I added mysql as co-maintainer and I squashed some commits (I hope I didn't make any mistake doing this)

File: net-analyzer/gvmd/Manifest
Comment (thesamesam) [inline net-analyzer/gvmd/Manifest:1]: The commit message here isn't very descriptive.. "to fix bug 21.4.5-r5" ok how? what was wrong with it?

Comment (thesamesam) [reply net-analyzer/gvmd/Manifest:1]: also, in this commit, you're adding a new distfile for 22.4.2 which I assume is a mistake

**Review by Flowdalic — CHANGES_REQUESTED** [review]
Thanks for your effort on GMV in Gentoo. Much appreciated. :)

Ideally also please rebase the commit history, so that, for example, fixup commits are merged into the commit they fix.

Since this is already a pretty large PR, please also consider removing changes where possible. For example, I think you do not have to add gvmd-21.4.5-r6 if you also add gvmd-22.5.0. Same is probably true for gsad-21.4.4-r1

File: net-analyzer/python-gvm/python-gvm-23.5.1.ebuild
Comment (Flowdalic) [inline net-analyzer/python-gvm/python-gvm-23.5.1.ebuild:26]: Tests fails for me with

```
ERROR: tests.xml.test_pretty_print (unittest.loader._FailedTest.tests.xml.test_pretty_print)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.xml.test_pretty_print
Traceback (most recent call last):
  File "/usr/lib/python3.11/unittest/loader.py", line 407, in _find_test_path
    module = self._get_module_from_name(name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/loader.py", line 350, in _get_module_from_name
    __import__(name)
  File "/data-scratch/var-tmp/portage/net-analyzer/python-gvm-23.5.1/work/python-gvm-23.5.1/tests/xml/test_pretty_print.py", line 25, in <module>
    from pontos.testing import temp_directory
ModuleNotFoundError: No module named 'pontos'
```

haven't looked into it yet.

File: net-analyzer/greenbone-feed-sync/greenbone-feed-sync-23.6.0.ebuild
Comment (Flowdalic) [inline net-analyzer/greenbone-feed-sync/greenbone-feed-sync-23.6.0.ebuild:63]: I think this should be enough

```suggestion
	if systemd_is_booted; then
```

File: net-analyzer/greenbone-feed-sync/greenbone-feed-sync-23.6.0.ebuild
Comment (Flowdalic) [inline net-analyzer/greenbone-feed-sync/greenbone-feed-sync-23.6.0.ebuild:56]: Only show the messages once
```suggestion
pkg_postinst() {
    if [[ -n ${REPLACING_VERSIONS} ]]; then
    	return
   fi
```

File: net-analyzer/greenbone-feed-sync/greenbone-feed-sync-23.6.0.ebuild
Comment (Flowdalic) [inline net-analyzer/greenbone-feed-sync/greenbone-feed-sync-23.6.0.ebuild:15]: I think this is GPL-3+ only?

```suggestion
LICENSE="GPL-3+"
```

File: net-analyzer/greenbone-feed-sync/files/greenbone-feed-sync.service
Comment (Flowdalic) [inline net-analyzer/greenbone-feed-sync/files/greenbone-feed-sync.service:9]: See the other comment regarding Group=gvm.

File: dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild
Comment (Flowdalic) [inline dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild:63]:
```suggestion
	if use test; then
```

File: dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild
Comment (Flowdalic) [inline dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild:50]:
```suggestion
	if $use test; then
```

File: net-analyzer/ospd-openvas/ospd-openvas-22.5.1.ebuild
Comment (Flowdalic) [inline net-analyzer/ospd-openvas/ospd-openvas-22.5.1.ebuild:41]: Drop this explicit declaration of `src_prepare` that only calls `default`.

File: net-analyzer/notus-scanner/notus-scanner-22.5.0.ebuild
Comment (Flowdalic) [inline net-analyzer/notus-scanner/notus-scanner-22.5.0.ebuild:38]: Drop this explicit declaration of `src_prepare` that only calls `default`.

File: net-analyzer/gvm-tools/gvm-tools-23.4.0.ebuild
Comment (Flowdalic) [inline net-analyzer/gvm-tools/gvm-tools-23.4.0.ebuild:25]: I think you should be able to drop the explicit definition of src_prepare here.

File: dev-db/pg-gvm/pg-gvm-22.5.1.ebuild
Comment (Flowdalic) [inline dev-db/pg-gvm/pg-gvm-22.5.1.ebuild:22]:
```suggestion
	>=dev-libs/libical-1.0.0:=
```

File: dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild
Comment (Flowdalic) [inline dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild:25]: Drop
```suggestion
```

File: dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild
Comment (Flowdalic) [inline dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild:51]:
```suggestion
		mv "${WORKDIR}"/"${MY_TEST_UTILS}"-"${MY_TEST_COMMIT}" "${WORKDIR}"/"${MY_TEST_UTILS}" || die
```

File: dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild
Comment (Flowdalic) [inline dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild:10]: Personally, I would consider declaring those as local variables in src_configure.

File: dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild
Comment (Flowdalic) [inline dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild:53]:
```suggestion
		mv "${WORKDIR}"/"${MY_TEST_UTILS}" "${BUILD_DIR}"/test/ || die
```

File: dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild
Comment (Flowdalic) [inline dev-libs/paho-mqtt-c/paho-mqtt-c-1.3.12.ebuild:54]:
```suggestion
		cp "${S}"/test/*.py "${BUILD_DIR}"/test/ || die
```

**Review by Flowdalic — CHANGES_REQUESTED** [review]
Thanks for your contribution. This PR approaches completion.

Please find my comments inline.

**Comment by juippis** [comment]
Thank you for this juggernaut of a task :)

**Pull request closed without merging**
